pkg/netflow/flowaggregator/flowaccumulator.go

```diff
@@ -85,15 +94,25 @@ func (f *flowAccumulator) flush() []*common.Flow {
 }
 
 func (f *flowAccumulator) add(flowToAdd *common.Flow) {
-	f.mu.Lock()
-	defer f.mu.Unlock()
+	log.Tracef("Add new flow: %+v", flowToAdd)
+
+	// Handle port rollup
+	f.portRollup.Add(flowToAdd.SrcAddr, flowToAdd.DstAddr, uint16(flowToAdd.SrcPort), uint16(flowToAdd.DstPort))
+	portCount, isEphemeralSource := f.portRollup.GetPortCount(flowToAdd.SrcAddr, flowToAdd.DstAddr, uint16(flowToAdd.SrcPort), uint16(flowToAdd.DstPort))
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
Rather then returning a port count and a boolean, I would return a sourcePortCount and a destinationPortCount. Or, even better, return two booleans, or an enum, telling if this is an ephemeralSourcePort, an ephemeralDestPort, or a non ephemeral conversation. To me `portRollup` should handle the logic of the threshold, sounds like you are mixing business boundaries here

Reply:
Fixed here https://github.com/DataDog/datadog-agent/pull/13367/commits/168af5b98ab357b7fbce83fc959a9a41280df616